plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/LazyConnectionFactory.java

```diff
@@ -40,7 +40,6 @@ public LazyConnectionFactory(@ForLazyConnectionFactory ConnectionFactory delegat
 
     @Override
     public Connection openConnection(ConnectorSession session)
-            throws SQLException
```

Comment:
Keeping this helps update the code in the future, would i want to throw a checked exception (as super/interface allows me to)

Reply:
If you only call this method via the interface, you should be safe

Reply:
i mean a future self who will be changing the implementation.

Reply:
Then your future self will add this exception back? It doesn't affect the interface, or its callers, only the override.

Reply:
Yes, except

- except the method is not allowed to throw a checked exception (i need to consult parent for that, which i may not do), so i may e.g. wrap IOException in Runtime unnecessarily and reviewer may miss that either 
- someone can call the impl directly (not via interface), so such legit reintroduction of a checked exception may still be a breaking change for callers

Reply:
I'm not sure that, when confronted with a checked exception, the first impulse is to wrap it instead of trying to rethrow (at which point the compiler will tell you if it's allowed by the interface). Likewise, I'm not convinced that we're passing the implementation type around as static type all that much.

But I agree these are all valid concerns.

We can always declare that this check is too immature to be enabled. For me it's not aggressive enough, but on the other hand it will always have cases like this, which can be considered overeager and false positives.

Anyway, it's good to discuss this :)

Reply:
Should we suppress it?

Reply:
I suppressed the error in this method

Reply:
is this considered an errorprone bug and reported upstream?

Reply:
I'm personally still on the fence on this. On the one hand I'd love it to lead to removal of dead code more (if none of the implementations throw this, we can remove the `throws` from the interface declaration!), but on the other I guess you're right that it's better to be conservative here.

I can create an issue upstream to see what they think.

Reply:
google/error-prone#3425